test/canary/canary_test.go

```diff
@@ -32,46 +30,83 @@ func init() {
 	environment.RegisterEnvironmentMetaDataFlags(envMetaDataStrings)
 }
 
-// This copies the canary config, starts the agent, and confirm version is correct.
-// Sanity test runs earlier in the terraform code execution
+// TestCanary verifies customers can download the agent, install it, and it "works".
+// Reports metrics on the number of download failures, install failures, and start failures.
 func TestCanary(t *testing.T) {
-	// Canary set up
-	downloadVersionFile()
+	installerFilePath := "./downloaded_cwa.rpm"
+	err := downloadInstaller(installerFilePath)
+	reportMetric(t, "DownloadFail", err)
+
+	// Don't care if uninstall fails. Agent might not be installed anyways.
+	_ = common.UninstallAgent()
+
+	err = common.InstallAgent(installerFilePath)
+	reportMetric(t, "InstallFail", err)
+
 	common.CopyFile(configInputPath, configOutputPath)
-	common.StartAgent(configOutputPath, true)
+	err = common.StartAgent(configOutputPath, false)
+	reportMetric(t, "StartFail", err)
 
-	// Version validation
-	expectedVersion, err := os.ReadFile(downloadAgentVersionPath)
-	if err != nil {
-		t.Fatalf("Failure reading downloaded version file %s err %v", downloadAgentVersionPath, err)
-	}
-	actualVersion, err := os.ReadFile(installAgentVersionPath)
-	if err != nil {
-		t.Fatalf("Failure reading installed version file %s err %v", installAgentVersionPath, err)
+	actualVersion, _ := os.ReadFile(installAgentVersionPath)
+	expectedVersion, _ := getVersionFromS3()
+	if expectedVersion != string(actualVersion) {
+		err = errors.New("agent version mismatch")
 	}
-	assert.Equal(t, string(expectedVersion), string(actualVersion))
+	reportMetric(t, "VersionFail", err)
+
+	setupCron()
 }
 
-func downloadVersionFile() {
-	metadata := environment.GetEnvironmentMetaData(envMetaDataStrings)
-	ctx := context.Background()
-	c, err := config.LoadDefaultConfig(ctx)
+// reportMetric is just a helper to report a metric and conditionally fail
+// the test based on the error.
+func reportMetric(t *testing.T, name string, err error) {
+	var v float64 = 0
```

Comment:
I thought of doing that, but my general preference is simple code and explicit code. Sometimes those things contradict though. In this case I felt like it doesn't hurt to explicitly write the initial value, and it makes it more "readable" to devs, especially those not super familiar with Golang. Just my opinion though... If we have LINT rules that say otherwise I will happily conform. Or if anyone else +1 your comment :)

Reply:
I'm indifferent. I will say that if I just saw `v := 0`, I will assume it is an int.